server/controllers/medical/patients/index.js

```diff
@@ -437,6 +437,12 @@ function find(options) {
   filters.dateTo('dateBirthTo', 'dob');
   filters.equals('health_zone');
   filters.equals('health_area');
+
+  // filters for location
+  const orignSql = `(originVillage.name LIKE ?) OR (originSector.name LIKE ?) OR (originProvince.name LIKE ?)`;
+  let village = options.originVillageName || ``;
+  village = ''.concat('%', village, '%');
+  filters.customMultiParameters('originVillageName', orignSql, [village, village, village]);
```

Comment:
This parameter can be more than just the village, correct?  Would it make more sense to call it "originLocationLabel" or something of that nature?

Reply:
the server filter service  cached when I used a name that doesn't exist in the sql result, that is why I was obliged to use originVillage

Reply:
I believe the trick is to make sure you have the same name in three places:
 1. The client-side filter service [here](https://github.com/IMA-WorldHealth/bhima-2.X/pull/2658/files#diff-a1aef484cc7f9a5f88d0d8555acc93fdR223)
 2. The "searchQueries" array in the search modal [here](https://github.com/IMA-WorldHealth/bhima-2.X/pull/2658/files#diff-c12093d0d555baf8f446bd24643f1dd1R26)
 3. The server-side label that is passed to FilterParser [here](https://github.com/IMA-WorldHealth/bhima-2.X/pull/2658/files#diff-4737dbdec30279777bd8d18c425035f8R445).

We have used non-column names before for our filters.  If you run into trouble, you can check out the [JournalService's filters](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/journal/journal.service.js#L117) which includes the non-column filters `includeNonPosted` and `hrEntity`.